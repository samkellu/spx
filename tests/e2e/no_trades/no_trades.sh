#! /bin/bash
RED='\033[0;31m'
GREEN='\033[0;34m'
YELLOW='\033[0;33m'
NC='\033[0m'

test_name="no_trades"

gcc tests/E2E/$test_name/test_trader.c -o tests/E2E/$test_name/test_trader -Wall -Werror -Wvla -O0 -std=c11 -g -fsanitize=address,leak -D_POSIX_C_SOURCE=199309L -lm

./spx_exchange "products.txt" "tests/E2E/$test_name/test_trader" | tee tests/E2E/$test_name/$test_name.test

diff tests/E2E/$test_name/$test_name.out tests/E2E/$test_name/$test_name.test && echo -e "${GREEN}Test $test_name passed.${NC}" || echo -e "${RED}Test $test_name failed.${NC}"

gcov ./spx_exchange

rm tests/E2E/$test_name/$test_name.test
