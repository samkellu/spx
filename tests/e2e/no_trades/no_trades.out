[SPX] Starting
[SPX] Trading 2 products: GPU Router
[SPX] Created FIFO /tmp/spx_exchange_0
[SPX] Created FIFO /tmp/spx_trader_0
[SPX] Starting trader 0 (tests/E2E/no_trades/test_trader)
[SPX] Connected to /tmp/spx_exchange_0
[SPX] Connected to /tmp/spx_trader_0
[SPX] Trader 0 disconnected
[SPX] Trading completed
[SPX] Exchange fees collected: $0
